src/libmongoc/tests/test-mongoc-client-pool.c

```diff
@@ -384,9 +427,14 @@ test_client_pool_install (TestSuite *suite)
    TestSuite_Add (
       suite, "/ClientPool/handshake", test_mongoc_client_pool_handshake);
 
+   TestSuite_Add (suite,
+                  "/ClientPool/create_client_pool_unused_session",
+                  test_client_pool_create_unused_session);
```

Comment:
Ah I missed this initially. `TestSuite_Add` adds this test to always run. This test requires running against a live server that supports sessions.

That is the cause of task failures on Evergreen like [this one](https://evergreen.mongodb.com/task/mongo_c_driver_asan_ubuntu_debug_compile_asan_clang_patch_43c5aaf21185156cc9e56ac25ac105c51f0a44be_5ef2622b2fbabe7b1c89a22e_20_06_23_20_12_29):
```
[2020/06/23 20:14:23.716] Begin /ClientPool/create_client_pool_unused_session, seed 2599939822
[2020/06/23 20:14:33.220] error calling ismaster: 'No suitable servers found: `serverselectiontimeoutms` timed out: [connection refused calling ismaster on 'localhost:27017']'
[2020/06/23 20:14:33.220] URI = mongodb://localhost:27017
```
The test is attempting to run, but the test runner in the compile task is meant to only run tests that do not require a live server (mock server tests and unit tests).

I believe this should do the trick:
```
TestSuite_AddFull (suite,
                    "/ClientPool/create_client_pool_unused_session",
                    test_client_pool_create_unused_session,
                    NULL /* dtor */,
                    NULL /* ctx */,
                    test_framework_skip_if_no_sessions);
```

You can verify this locally by running the test without a server running. The test should skip instead of failing.